wnfs-common/src/link.rs

```diff
@@ -125,14 +125,11 @@ impl<T: RemembersCid> Link<T> {
         T: DeserializeOwned,
     {
         match self {
-            Self::Encoded {
-                ref cid,
-                value_cache,
-            } => match value_cache.into_inner() {
+            Self::Encoded { cid, value_cache } => match value_cache.into_inner() {
                 Some(cached) => Ok(cached),
                 None => {
-                    let value: T = store.get_deserializable(cid).await?;
-                    value.persisted_as().get_or_init(async { *cid }).await;
+                    let value: T = store.get_deserializable(&cid).await?;
+                    value.persisted_as().get_or_init(async { cid }).await;
```

Comment:
We probably need the same fix here.

If we're in `Self::Encoded`, then resolve the `value_cache` (or load the cid), and then set `*self = Self::Decoded`, before returning `&mut T`.

Reply:
Can you suggest a fix for this?

Reply:
```rs
pub async fn resolve_value_mut(&mut self, store: &(impl BlockStore + ?Sized)) -> Result<&mut T>
where
    T: DeserializeOwned,
{
    match self {
        Self::Encoded { cid, value_cache } => {
            let value = match value_cache.take() {
                Some(v) => v,
                None => {
                    let value: T = store.get_deserializable(cid).await?;
                    value.persisted_as().get_or_init(async { *cid }).await;
                    value
                }
            };

            *self = Self::Decoded { value };

            Ok(match self {
                Self::Encoded { .. } => unreachable!(),
                Self::Decoded { value } => value,
            })
        }
        Self::Decoded { value, .. } => Ok(value),
    }
}
```

Reply:
Yeah, you were probably referring to `resolve_value_mut` and not `resolve_owned_value`